org-roam-capture.el

```diff
@@ -277,25 +277,54 @@ This function is used solely in Org-roam's capture templates: see
     (widen)
     (goto-char (point-max))))
 
+(define-error 'org-roam-malformed-template
+  "Malformed template in `org-roam-capture-templates'"
+  'org-roam-errors)
+
+(defun org-roam-capture--validate-template (template)
+  "Validate TEMPLATE by checking its elements.
+Return t if the TEMPLATE is well-formed, nil otherwise.
+As a special case, return 'group if TEMPLATE is a template-group.
+See `org-roam-capture-templates' for details."
+  (pcase template
+    (`(,(pred stringp) ,(pred stringp))
+     'group)
+    ((pred (lambda (x)
+             (>= (length x) 5)))
+     (pcase (cl-subseq template 0 5)
+       (`(,(pred stringp)
+          ,(pred stringp)
+          ,(pred symbolp)
+          ,(pred listp)
+          ,(pred stringp))
+        t)
```

Comment:
I really thought that `_` would save me from having to `cl-subseq` the list, i.e., by only considering the the first 5 members of the list, but it didn't.  I actually think adding a `_*` expansion in `pcase` would be useful.

…Or maybe I'm just missing something really obvious.